To-do-list/AppDelegate.swift

```diff
@@ -11,40 +11,57 @@ import UIKit
 class AppDelegate: UIResponder, UIApplicationDelegate {
 
     var window: UIWindow?
+    var storyboard = UIStoryboard(name: "Main", bundle: nil)
+    var navigationController: UINavigationController?
     
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
       
-        let storyboard = UIStoryboard(name: "Main", bundle: nil)
-        guard let controller = storyboard.instantiateViewController(withIdentifier: "ViewController") as? ViewController else { return false }
-        let mainViewModel = MainViewModel()
+        let controller = createViewContorller()
+        navigationController = UINavigationController(rootViewController: controller)
         
-        guard let taskController = storyboard.instantiateViewController(withIdentifier: "TaskViewController") as? TaskViewController else { return false }
-        let taskViewModel = TaskViewModel()
+        window = UIWindow(frame: UIScreen.main.bounds)
+        window?.rootViewController = navigationController
+        window?.makeKeyAndVisible()
         
-        let navigationController = UINavigationController(rootViewController: controller)
-        mainViewModel.onOpenTaskViewController = { [ weak self ] in
-            navigationController.pushViewController(taskController, animated: true)
+        return true
+    }
+    
+    func createViewContorller() -> UIViewController {
+        guard let controller = storyboard.instantiateViewController(withIdentifier: "ViewController") as? ViewController else { return .init() }
+        let viewModel = ViewModel()
+        viewModel.onOpenTaskViewController = { [ weak self ] in
+            guard let self = self else { return }
+            let taskController = self.createTaskViewController(controller: controller)
+            self.navigationController?.pushViewController(taskController, animated: true)
         }
-        controller.configure(mainViewModel: mainViewModel)
-        
+        viewModel.onOpenEditViewController = { [ weak self ] in
+            guard let self = self else { return }
+            let taskModel = TaskViewModel()
```

Comment:
where did you set taskModel to your taskController?
try to create function for this logic

Reply:
I set a function configureTaskModel()